Added Annotation volume.alpha.kubernetes.io/node-affinity (#42340)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

set from the command line flag (`--node-ip`). This IP is verified with the cloud provider as valid
by the cloud-controller-manager.

> ### volume.alpha.kubernetes.io/node-affinity (alpha) {#alpha-kubernetes-io-provided-node-ip}

Reviewer: ```suggestion
### volume.alpha.kubernetes.io/node-affinity (alpha) {#volume-alpha-kubernetes-io-node-affinity}
```

Author: Done @sftim

Type: Annotation

Example: `volume.alpha.kubernetes.io/node-affinity: "constrain-with-nodes"`

Used on: Node

> Node affinity and [nodeSelector](https://kubernetes.io/docs/reference/labels-annotations-taints/#schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling of your Pod to specific nodes in the cluster, based on node labels.

Reviewer 2: We prefer to use site-relative links whenever feasible.
```suggestion
Node affinity and [nodeSelector](/docs/reference/labels-annotations-taints/#schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling of your Pod to specific nodes in the cluster, based on node labels.
```

Author: Hey @dipesh-rawat, I added the address you gave but it's giving me `page not found`.
So here's what I did I added `https://github.com/kubernetes/website/blob/main/content/en/docs/reference/labels-annotations-taints/_index.md#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector` for now.
So could you assist me with this page not found issue with the address you gave?

Reviewer 3: This link works fine for me, can you recheck it?
This won't work in IDE and has to be tested after running the website locally using `make serve`

Author: > This link works fine for me, can you recheck it? This won't work in IDE and has to be tested after running the website locally using `make serve`

Yaa @Ritikaa96,  after checking it locally the address doesn't work so for now I have added the link!

> See [node affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) for more information. 

Reviewer 2: ```suggestion
See [node affinity](/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) for more information. 
```

Author: Same with this `/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity`, when I add this with the node affinity it's redirecting me to a `page not found`.
So may I add this address to the `node affinity` `https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/scheduling-eviction/assign-pod-node.md/#node-affinity` ?

> For more details, see [node affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) kubernets docs `#node-affinity`.

Author: Hi @sftim, I have made the changes.

set from the command line flag (`--node-ip`). This IP is verified with the cloud provider as valid
by the cloud-controller-manager.

### volume.alpha.kubernetes.io/node-affinity (alpha) {#volume-alpha-kubernetes-io-node-affinity}

Type: Annotation

Example: `volume.alpha.kubernetes.io/node-affinity: "{}"`

Used on: PersistentVolume

> The `volume.alpha.kubernetes.io/node-affinity` annotation was used to specify experimental node affinity
> rules for PersistentVolumes (PV) in Kubernetes. Volume _[node affinity](/docs/concepts/storage/persistent-volumes/#node-affinity)_ allows you to restrict the
> scheduling of Pods that use the corresponding PersistentVolume so that the Pods can only bound to
> appropriate nodes in your cluster, based on the {{<glossary_tooltip text="labels" term_id="label">}}
> that are set on those nodes.

Reviewer 4: ```suggestion
The `volume.alpha.kubernetes.io/node-affinity` annotation was used to specify 
experimental node affinity rules for PersistentVolumes (PV) in Kubernetes.
Volume _[node affinity](/docs/concepts/storage/persistent-volumes/#node-affinity)_
allows you to restrict the scheduling of Pods that use the corresponding PersistentVolume
so that the Pods can only bound to appropriate nodes in your cluster, based on
the {{<glossary_tooltip text="labels" term_id="label">}} that are set on those Nodes.
```

set from the command line flag (`--node-ip`). This IP is verified with the cloud provider as valid
by the cloud-controller-manager.

### volume.alpha.kubernetes.io/node-affinity (alpha) {#volume-alpha-kubernetes-io-node-affinity}

Type: Annotation

Example: `volume.alpha.kubernetes.io/node-affinity: "{}"`

Used on: PersistentVolume

> The `volume.alpha.kubernetes.io/node-affinity` annotation was used to specify 

Reviewer 4: ```suggestion
The `volume.alpha.kubernetes.io/node-affinity` annotation is used to specify 
```

experimental node affinity rules for PersistentVolumes (PV) in Kubernetes.
Volume _[node affinity](/docs/concepts/storage/persistent-volumes/#node-affinity)_
allows you to restrict the scheduling of Pods that use the corresponding PersistentVolume

> so that the Pods can only bound to appropriate nodes in your cluster, based on

Reviewer 4: The logic is confusing ...
I'm supposed to set this on a PV to impact the scheduling of a Pod?
Sounds that implies an annotation on the Pod rather than the PV, right?
I'm not familiar with the background, so forgive me about the dumb questions.

set from the command line flag (`--node-ip`). This IP is verified with the cloud provider as valid
by the cloud-controller-manager.

### volume.alpha.kubernetes.io/node-affinity (alpha) {#volume-alpha-kubernetes-io-node-affinity}

Type: Annotation

> Example: `volume.alpha.kubernetes.io/node-affinity: "constrain-with-nodes"`

Reviewer: ```suggestion
Example: `volume.alpha.kubernetes.io/node-affinity: "{}"`
```
As the field no longer works, I wouldn't use a complex example.

If we want one, it could look like:
 `volume.alpha.kubernetes.io/node-affinity: '{"requiredDuringSchedulingIgnoredDuringExecution":{"nodeSelectorTerms":[{"matchExpressions":[{"key":"kubernetes.io/hostname","operator":"In","values":["example-node"]}]}]}}'"`

> Used on: Node

Reviewer: ```suggestion
Used on: PersistentVolume
```

> The `volume.alpha.kubernetes.io/node-affinity` annotation is used to specify node affinity rules for a 
> PersistentVolume (PV) in Kubernetes. It allows you to restrict the scheduling of Pods that use the corresponding 
> PersistentVolume to specific nodes in the cluster, based on [node labels](/docs/concepts/storage/persistent-volumes/#selector).

Reviewer: ```suggestion
The `volume.alpha.kubernetes.io/node-affinity` annotation was used to specify experimental node affinity
rules for PersistentVolumes (PV) in Kubernetes. Volume _node affinity_ allows you to restrict the
scheduling of Pods that use the corresponding PersistentVolume so that the Pods can only bound to
appropriate nodes in your cluster, based on the {{<glossary_tooltip text="labels" term_id="label">}}
that are set on those nodes.

You should now use the [`nodeAffinity` field](/docs/concepts/storage/persistent-volumes/#node-affinity) instead.
```

> Node affinity and [nodeSelector](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/labels-annotations-taints/_index.md#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling 

Reviewer 5: ```suggestion
Node affinity and [nodeSelector](/reference/labels-annotations-taints/_index.md#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling 
```

> Node affinity and [nodeSelector](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/labels-annotations-taints/_index.md#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling 

Reviewer 5: ```suggestion
Node affinity and [nodeSelector](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/labels-annotations-taints/_index.md#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector) are conceptually similar, both letting you restrict the scheduling 
```
 Nit

> Node affinity and [nodeSelector](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/labels-annotations-taints/_index.md#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling 

Reviewer 5: Make sure that you're wrapping lines at 80 characters for ease of reviewing (unless a link URL exceeds the 80 chars which is fine)

> Node affinity and [nodeSelector](/docs/reference/labels-annotations-taints/#scheduleralphakubernetesionode-selector-schedulerkubernetesnode-selector) are conceptually similar, both enabling you to restrict the scheduling 
> of your Pod to specific nodes in the cluster, based on node labels.
> 
> For more details, see [node affinity](/docs/concepts/storage/persistent-volumes/#node-affinity)
> .

Reviewer: ```suggestion
```

> in the documentation about PersistentVolumes.

Reviewer 5: ```suggestion
.
```
